### ISSUE TITLE
Complex form control broken Accessibility

### ISSUE THREAD

**Issue by krshna90** [ant-design/ant-design#53431]
### Reproduction link

[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/p/devbox/jolly-leaf-7vz597)

### Steps to reproduce

1. Create a complex form control as per the documentation:
[Form](https://ant.design/components/form)
  ```
<Form.Item label="Field">
    <Form.Item name="field" noStyle><Input /></Form.Item> // that will bind input
    <span>description</span>
  </Form.Item>
```

2. Click on the label / try a screen reader

### What is expected?

The form control should be activated on click of its associated label.

### What is actually happening?

Nothing happens. The association of Form Label and Form Item is broken.

| Environment | Info |
| --- | --- |
| antd | 5.16.2 |
| React | 18.2.14 |
| System | Mac 15.4 |
| Browser | Chrome 134 |

---
It does work when `htmlFor={"field"}` is added to the parent   `<Form.Item label="Field">`.
However it is expected to work out of box similar to a simple form control. Also there is not documentation confirming this solution.

**Comment by krshna90** [comment]
hey @afc163 yes `htmlFor` works. I had already tried it but its hard to align with everyone when the documentation does not mention it.
Also, I believe it should be out of box since I followed the complex form control suggested in the documentation

**Comment by afc163** [comment]
@krshna90 PR is welcome~

**Comment by nrps9909** [comment]
I can work on the documentation fix: update the official complex-form-control demo to explicitly pair the outer Form.Item htmlFor with the nested control id, document why this is needed, and add a focused accessibility regression test. I will keep the change scoped to the single-control-with-adjacent-content case.
